src/main/java/uk/gov/beis/els/categories/spaceheaters/HeatPumpCombinationHeatersFormSequenceProvider.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.beis.els.categories.spaceheaters;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.hibernate.validator.spi.group.DefaultGroupSequenceProvider;
+import uk.gov.beis.els.categories.spaceheaters.model.HeatPumpCombinationHeatersForm;
+import uk.gov.beis.els.categories.spaceheaters.model.SoundPowerLevelIndoorsGroup;
+
+public class HeatPumpCombinationHeatersFormSequenceProvider implements DefaultGroupSequenceProvider<HeatPumpCombinationHeatersForm> {
```

Comment:
I'm a bit confused as to why these new sequence provider and the `SoundPowerLevelIndoorsGroup` are required. From a quick look this seems to add the `SoundPowerLevelIndoorsGroup` which is only on the NotNull check on the indoor sound power level field. So it seems to add the NotNull check only if the field is already known to be not null?

As a quick test I removed this sequence provider from the `HeatPumpCombinationHeatersForm` form and removed `@NotNull(groups = SoundPowerLevelIndoorsGroup.class)` from the `soundPowerLevelIndoors` field, and both the HTML forms and the API seemed to behave as expected. i.e. field not provided = valid. field provided but with value 999 = invalid, too many digits.

Is there a particular error or behaviour you were seeing that needed this? 
Same with the boiler combination heaters

Reply:
@jamesbarnett91 I added the sequence provider and group because I remembered we have to add `@NotNull` to all fields in the form. And so if I added it without the group it will always required it.
Unless I dont need the `@NotNull` check.

Reply:
done ;)

Reply:
Thanks 👍